metrics.py

```diff
@@ -114,23 +112,31 @@ def get_resource_metrics_for_executor_tasks(tasks):
 
     cpu_millis = (last_task_to_finish.end_total_cpu_jiffies -
       first_task_to_start.start_total_cpu_jiffies) * MILLIS_PER_JIFFY
-    total_transmit_idle_millis = (last_task_to_finish.end_network_transmit_idle_millis -
+    transmit_idle_millis = (last_task_to_finish.end_network_transmit_idle_millis -
       first_task_to_start.start_network_transmit_idle_millis)
-    network_metrics = NetworkMetrics(elapsed_millis, total_transmit_idle_millis)
+    network_metrics = NetworkMetrics(
+      elapsed_millis=elapsed_millis,
+      transmit_idle_millis=transmit_idle_millis)
 
     disk_name_to_metrics = {}
     for disk_name, disk_utilization in first_task_to_start.disk_utilization.iteritems():
       disk_name_to_metrics[disk_name] = DiskMetrics(
-        elapsed_millis,
-        first_task_to_start.disk_utilization[disk_name].start_counters,
-        last_task_to_finish.disk_utilization[disk_name].end_counters)
+        elapsed_millis=elapsed_millis,
+        start_counters=first_task_to_start.disk_utilization[disk_name].start_counters,
+        end_counters=last_task_to_finish.disk_utilization[disk_name].end_counters)
 
     gc_millis = last_task_to_finish.end_gc_millis - first_task_to_start.start_gc_millis
-    return ExecutorResourceMetrics(len(tasks), elapsed_millis, cpu_millis, network_metrics,
-      disk_name_to_metrics, gc_millis)
+    return ExecutorResourceMetrics(
```

Comment:
As I'm looking at this again, the best way to write this IMO would have been to have all of this code in the ExecutorResourceMetrics constructor. Don't fix this now!! Just design food-for-thought for the future.